File: python/python/Java/PythonParserBase.java

import org.antlr.v4.runtime.Parser;
import org.antlr.v4.runtime.TokenStream;

public abstract class PythonParserBase extends Parser
{
    public PythonVersion Version = PythonVersion.Autodetect;

    protected PythonParserBase(TokenStream input) {
        super(input);
    }

    protected boolean CheckVersion(int version) {
        return Version == PythonVersion.Autodetect || version == Version.getValue();
    }

    protected void SetVersion(int requiredVersion) {
        if (requiredVersion == 2) {
            Version = PythonVersion.Python2;
        } else if (requiredVersion == 3) {
            Version = PythonVersion.Python3;
        }
    }
}

File: python/python/CSharp/PythonParserBase.cs
using Antlr4.Runtime;
using System.IO;

public enum PythonVersion
{
    Autodetect,
    Python2 = 2,
    Python3 = 3
}

public abstract class PythonParserBase : Parser
{
    public PythonVersion Version { get; set; }

    protected PythonParserBase(ITokenStream input) : base(input)
    {
    }

    protected PythonParserBase(ITokenStream input, TextWriter output, TextWriter errorOutput)
        : base(input, output, errorOutput)
    { }

    protected bool CheckVersion(int version) => Version == PythonVersion.Autodetect || version == (int) Version;

    protected void SetVersion(int requiredVersion) => Version = (PythonVersion) requiredVersion;
}

